teuthology/orchestra/opsys.py

```diff
@@ -85,7 +86,7 @@ def __init__(self, name=None, version=None, codename=None):
     @staticmethod
     def _version_to_codename(name, version):
         for (_version, codename) in DISTRO_CODENAME_MAP[name].iteritems():
-            if version == _version or version.split('.')[0] == _version:
+            if str(version) == _version or str(version).split('.')[0] == _version:
```

Comment:
Why the change to stringify variables here? Aren’t they already strings?

Reply:
> Why the change to stringify variables here? Aren’t they already strings?

At the time of patch applied, the version looks like 'X.Y' and it appeared here as floating, so... this code made sure that there is a string.